kliff/trainer/base_trainer.py

```diff
@@ -0,0 +1,768 @@
+        random.seed(self.workspace["seed"])
+        # torch.manual_seed(self.seed) # enable torch seed in respective children
+        # torch.cuda.manual_seed_all(self.seed)
+        # torch.backends.cudnn.deterministic = True
+        # torch.backends.cudnn.benchmark = False
+
+    def setup_workspace(self):
+        """
+        Check all the existing runs in the root directory and see if it finished the run
+        or not. If it is finished, it will start a new run. If it is not finished, it will
+        resume the training. If the resume is not requested, it will start a new run.
+        """
+        dir_list = sorted(
+            glob(f"{self.workspace['name']}/{self.model_manifest['name']}*")
+        )
+        dir_list = [p for p in dir_list if os.path.isdir(p)]
+
+        if len(dir_list) == 0 or not self.workspace["resume"]:
+            logger.info(
+                "Either a fresh run or resume is not requested. Starting a new run."
+            )
+            self.current["appending_to_previous_run"] = False
+            self.current["run_dir"] = (
+                f"{self.workspace['name']}/{self.current['run_title']}"
+            )
+            os.makedirs(self.current["run_dir"], exist_ok=True)
+        else:
+            last_dir = dir_list[-1]
+            was_it_finished = os.path.exists(f"{last_dir}/.finished")
+            if was_it_finished:  # start new run
+                logger.warning(
+                    "Resuming from last training was requested, but it was completed. Exiting."
+                )
+                # current_run_dir = (
+                #     f"{self.workspace['name']}/{self.current['run_title']}"
+                # )
+                # os.makedirs(current_run_dir, exist_ok=True)
+                # self.current["appending_to_previous_run"] = False
+                sys.exit()
+            else:
+                logger.info("Last trainer was not finished. Resuming the training.")
+                self.current["appending_to_previous_run"] = True
+                self.current["run_dir"] = dir_list[-1]
+
+        # make dataset directory
+        self.current["data_dir"] = f"{self.workspace['name']}/datasets"
+        os.makedirs(self.current["data_dir"], exist_ok=True)
+
+    def setup_dataset(self):
+        """
+        Set up the dataset based on the provided information.
+        TODO: ColabFit integration for extreme scale datasets.
+        """
+        dataset_module_manifest = deepcopy(self.dataset_manifest)
+        dataset_module_manifest["weights"] = self.loss_manifest["weights"]
+        dataset_list = _parallel_read(
+            self.dataset_manifest["path"],
+            num_chunks=self.optimizer_manifest["num_workers"],
+            energy_key=self.dataset_manifest.get("keys", {}).get("energy", "energy"),
+            forces_key=self.dataset_manifest.get("keys", {}).get("forces", "forces"),
+        )
+        self.dataset = deepcopy(dataset_list[0])
```

Comment:
Because we want to merge the multiprocessed list of configurations into a single dataset. So I copied the fist one and extended it with remaining chunks. The list contains several datasets (no configurations) So I cant directly use it.